src/WhereConditions/WhereHasConditionsDirective.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Nuwave\Lighthouse\WhereConditions;
+
+class WhereHasConditionsDirective extends WhereConditionsBaseDirective
+{
+    public static function definition(): string
+    {
+        return /** @lang GraphQL */ <<<'SDL'
+"""
+Allows clients to filter a query based on the existence of a related model, using
+a dynamically controlled `WHERE` condition that applies to the relationship.
+"""
+directive @whereHasConditions(
+    """
+    The Eloquent relationship that the conditions will be applied to.
+    This argument can be ommited if the field name follows a `hasRelationship` naming convention.
+    Example: If the Eloquent relationship is named `posts`, you can name your field  `hasPosts` and
+    ommit this argument; the `posts` relationship will be automatically resolved.
+    """
+    relation: String
+
+    """
+    Restrict the allowed column names to a well-defined list.
+    This improves introspection capabilities and security.
+    """
+    columns: [String!]
+) on ARGUMENT_DEFINITION | INPUT_FIELD_DEFINITION
+SDL;
+    }
+
+    /**
+     * @param  \Illuminate\Database\Query\Builder|\Illuminate\Database\Eloquent\Builder  $builder
+     * @param  mixed[]  $whereConditions
+     * @return \Illuminate\Database\Query\Builder|\Illuminate\Database\Eloquent\Builder
+     */
+    public function handleBuilder($builder, $whereConditions)
+    {
+        // Make sure to ignore empty conditions.
```

Comment:
Why? That might still be a valid filter: Determine if any related model exists.

Reply:
I wanted to achieve the same behavior that you get when you supply an empty condition (`{}`) to the `@whereConditions` directive; it just has no effect on the query.

The reason is that when you write a client and a query in the schema has a `@whereConditions` arg and a `@whereHasConditions` arg, it's simpler to write a GraphQL query that uses the same "null" value for both args.